From: Arvind Yadav <arvind.yadav.cs@gmail.com>
Date: Sun, 16 Jul 2017 12:00:46 +0530
Subject: infiniband: pvrdma: constify pci_device_id.
Patch-mainline: v4.14-rc1
Git-commit: 7806def081b2185a488fb31bd1deb4f5a3fe4765
References: bsc#1046306 FATE#322942

pci_device_id are not supposed to change at runtime. All functions
working with pci_device_id provided by <linux/pci.h> work with
const pci_device_id. So mark the non-const structs as const.

File size before:
   text	   data	    bss	    dec	    hex	filename
  10774	   1872	      8	  12654	   316e infiniband/hw/vmw_pvrdma/pvrdma_main.o

File size After adding 'const':
   text	   data	    bss	    dec	    hex	filename
  10838	   1808	      8	  12654	   316e infiniband/hw/vmw_pvrdma/pvrdma_main.o

Signed-off-by: Arvind Yadav <arvind.yadav.cs@gmail.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/hw/vmw_pvrdma/pvrdma_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/infiniband/hw/vmw_pvrdma/pvrdma_main.c
+++ b/drivers/infiniband/hw/vmw_pvrdma/pvrdma_main.c
@@ -1077,7 +1077,7 @@ static void pvrdma_pci_remove(struct pci
 	pci_set_drvdata(pdev, NULL);
 }
 
-static struct pci_device_id pvrdma_pci_table[] = {
+static const struct pci_device_id pvrdma_pci_table[] = {
 	{ PCI_DEVICE(PCI_VENDOR_ID_VMWARE, PCI_DEVICE_ID_VMWARE_PVRDMA), },
 	{ 0 },
 };
